hail/src/main/scala/is/hail/expr/ir/LowerMatrixIR.scala

```diff
@@ -572,8 +572,15 @@ object LowerMatrixIR {
       case table => lowerChildren(table, ab).asInstanceOf[TableIR]
     }
 
-    if (lowered.typ != tir.typ)
-      fatal(s"lowering changed type:\n  before: ${ tir.typ }\n  after: ${ lowered.typ }")
+    assertTypeUnchanged(tir, lowered)
+    lowered
+  }
+
+  private[this] def lower(bmir: BlockMatrixIR, ab: ArrayBuilder[(String, IR)]): BlockMatrixIR = {
+    val lowered = bmir match {
+      case noMatrixChildren => lowerChildren(noMatrixChildren, ab).asInstanceOf[BlockMatrixIR]
```

Comment:
this is a trivial match statement. Just `val lowered = lowerChildren(bmir, ab)...`?

Reply:
Ya I thought it looked more consistent with the other lowerings and showed there just weren’t yet `BlockMatrixIR` that required a special pass. Happy to get rid of it if you want

Reply:
ok, sure.